curvefs/src/metaserver/storage/iterator.h

```diff
@@ -50,6 +50,8 @@ class Iterator {
 
     virtual std::string Value() = 0;
 
+    virtual void Value(const char** data, size_t* size) {}
```

Comment:
Can we just return `absl::string_view` and remove previous `virtual std::string Value() = 0`?

And also, `virtual std::string Key() = 0`

Reply:
I suppose returning string_view from RocksDBStorageIterator is ok, but I'm not sure whether returning string_view from Value() is safe for other derived iterators, since their returned string's life time may not be longer than this string_view.

Reply:
I think the only exception is https://github.com/opencurve/curve/blob/9de6b45672d68ebfa7fe43a57b83e9b267f51f59/curvefs/src/metaserver/storage/storage_fstream.h#L237, it returns `std::string` from a temporary.

Maybe we can store the temporary as member variable and returns string_view from it?

Reply:
> I think the only exception is
> 
> https://github.com/opencurve/curve/blob/9de6b45672d68ebfa7fe43a57b83e9b267f51f59/curvefs/src/metaserver/storage/storage_fstream.h#L237
> 
> , it returns `std::string` from a temporary.
> Maybe we can store the temporary as member variable and returns string_view from it?

Yes, we could use a member variable to store every string returned from InternalKey to endure its life time. 
But changing Iterator's Key() & Value() return type will involve changes to various upper layer logics, if this is acceptable, I  can start to modify these logics. @wu-hanqing

Reply:
> > I think the only exception is
> > https://github.com/opencurve/curve/blob/9de6b45672d68ebfa7fe43a57b83e9b267f51f59/curvefs/src/metaserver/storage/storage_fstream.h#L237
> > 
> > , it returns `std::string` from a temporary.
> > Maybe we can store the temporary as member variable and returns string_view from it?
> 
> Yes, we could use a member variable to store every string returned from InternalKey to endure its life time. But changing Iterator's Key() & Value() return type will involve changes to various upper layer logics, if this is acceptable, I can start to modify these logics. @wu-hanqing

I think it's acceptable.